tests/bmff-signing.test.ts

```diff
@@ -0,0 +1,135 @@
+import assert from 'node:assert/strict';
+import * as fs from 'node:fs/promises';
+import { X509Certificate } from '@peculiar/x509';
+import { BMFF } from '../src/asset';
+import { CoseAlgorithmIdentifier } from '../src/cose';
+import { SuperBox } from '../src/jumbf';
+import { BMFFHashAssertion, Manifest, ManifestStore, ValidationStatusCode } from '../src/manifest';
+import { LocalTimestampProvider } from '../src/rfc3161';
+
+const sourceFile = 'tests/fixtures/trustnxt-icon.heic';
+const targetFileV2 = 'tests/fixtures/trustnxt-icon-signed-v2-test.heic';
+const targetFileV3 = 'tests/fixtures/trustnxt-icon-signed-v3-test.heic';
+
+describe('BMFF Signing Tests', function () {
+    this.timeout(5000);
+
+    let manifestV2: Manifest | undefined;
+    let manifestV3: Manifest | undefined;
+
+    async function signAndVerify(version: 2 | 3) {
+        const targetFile = version === 2 ? targetFileV2 : targetFileV3;
+
+        // load the certificate
+        const x509Certificate = new X509Certificate(await fs.readFile('tests/fixtures/sample_es256.pem'));
+
+        // load and parse the private key
+        const privateKeyData = await fs.readFile('tests/fixtures/sample_es256.key');
```

Comment:
Since we now do this signing setup (reading the certificate, reading and parsing the key) in multiple tests, we should move it to a utility location.